x-pack/plugins/monitoring/server/lib/logs/get_log_types.ts

```diff
@@ -60,13 +60,27 @@ export async function getLogTypes(
     end,
   }: { clusterUuid?: string; nodeUuid?: string; indexUuid?: string; start: number; end: number }
 ) {
-  checkParam(filebeatIndexPattern, 'filebeatIndexPattern in logs/getLogTypes');
+  checkParam(logsIndexPattern, 'logsIndexPattern in logs/getLogTypes');
 
   const metric = { timestampField: '@timestamp' };
+
+  const typeFilter = {
+    bool: {
+      // The point here is to make the query work for `filebeat-*` index and with logs-* index pattern.
+      // `service.type` is not assigned by the filebeat in the agents.
+      // The filter by `data_stream.type` is hack to use `should` filter.
+      // 'data_stream.type' will always be 'logs' for `logs-*` indices.
+      should: [
+        { term: { 'service.type': 'elasticsearch' } },
+        { term: { 'data_stream.type': 'logs' } },
```

Comment:
The logic here has me a little concerned that this might break for standalone filebeat.

Should we maybe have a test that loads in es-archives for both filebeat-* and logs-* to ensure either can work (or even both to account for a transitional case?).

Reply:
Agreed, we've initially discussed that but ended up making sure `service.type` exists in datastreams as well. see https://github.com/elastic/integrations/pull/4033#discussion_r951950336